api/groups/transactionGroup.go

```diff
@@ -703,17 +717,42 @@ func (tg *transactionGroup) getTransactionsPoolNonceGapsForSender(sender string,
 	)
 }
 
-func invalidQuery(sender, fields string, lastNonce, nonceGaps bool) bool {
-	queryForSender := lastNonce || nonceGaps
-	if sender == "" && queryForSender {
-		return true
+func validateQuery(sender, fields string, lastNonce, nonceGaps bool) error {
+	if fields != "" && lastNonce {
+		return errors.ErrFetchingLatestNonceCannotIncludeFields
+	}
+
+	if fields != "" && nonceGaps {
+		return errors.ErrFetchingNonceGapsCannotIncludeFields
 	}
 
-	if fields != "" && queryForSender {
-		return true
+	if sender == "" && lastNonce {
+		return errors.ErrEmptySenderToGetLatestNonce
 	}
 
-	return false
+	if sender == "" && nonceGaps {
+		return errors.ErrEmptySenderToGetNonceGaps
+	}
+
+	if fields != "" {
+		return validateFields(fields)
+	}
+
+	return nil
+}
+
+func validateFields(fields string) error {
+	for _, c := range fields {
+		if c == ',' {
+			continue
+		}
+
+		if !unicode.IsLetter(c) || !unicode.IsLower(c) {
```

Comment:
as discussed, we could allow 'a' -> 'Z' so both `gasLimit` and `gaslimit` will work. you anway convert the string to lowercase.